packages/lib/src/interactions/box.ts

```diff
@@ -0,0 +1,61 @@
+import { Box3, Vector3 } from 'three';
+
+import type { Size } from '../vis/models';
+
+const ZERO_VECTOR = new Vector3(0, 0, 0);
+
+class Box extends Box3 {
+  public get size(): Size {
+    const { x: width, y: height } = this.getSize(new Vector3());
+    return { width, height };
+  }
+
+  public get center(): Vector3 {
+    return this.getCenter(new Vector3());
```

Comment:
Note that we should avoid accessing `box.center` and `box.size` multiple times in the same method. It is better to use destructuring: `const { center, size } = box;`.